src/meta_memcache/base/connection_pool.py

```diff
@@ -115,13 +115,19 @@ def get_connection(self) -> Generator[MemcacheSocket, None, None]:
         try:
             conn = self._pool.popleft()
         except IndexError:
+            conn = None
+
+        if conn is None:
             conn = self._create_connection()
```

Comment:
This change is to avoid the exception being so cryptic when an error connecting is chained on the IndexError and looks like a bug or something.